modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/RegionResourceImpl.java

```diff
@@ -111,22 +116,52 @@ public Page<Region> getRegionsPage(
 	public Region postCountryRegion(Long countryId, Region region)
 		throws Exception {
 
-		return _toRegion(
+		if (region.getTitle_i18n() == null) {
+			Map<String, String> titleMap = new HashMap<>();
+
+			for (Locale locale : _language.getAvailableLocales()) {
+				titleMap.put(_language.getLanguageId(locale), null);
+			}
+
+			region.setTitle_i18n(titleMap);
+		}
```

Comment:
Is it possible to extract this in a function? In this way we can reuse the same for both the methods without duplicate the code

Reply:
@smottal yep, I updated the commit, extracted into a private method _setTitleMap.